src/notification.c

```diff
@@ -129,22 +129,41 @@ void notification_run_script(struct notification *n)
                         int status;
                         waitpid(pid1, &status, 0);
                 } else {
+                        // second fork to prevent zombie processes
                         int pid2 = fork();
                         if (pid2) {
                                 exit(0);
                         } else {
-                                int ret = execlp(script,
+                                // Set environment variables
+                                gchar *n_id = g_strdup_printf("%i", n->id);
+                                gchar *n_progress = g_strdup_printf("%i", n->progress);
+                                gchar *n_timeout = g_strdup_printf("%li", n->timeout);
+                                gchar *n_timestamp = g_strdup_printf("%li", n->timestamp);
+                                safe_setenv("DUNST_APP_NAME",  appname);
+                                safe_setenv("DUNST_SUMMARY",   summary);
+                                safe_setenv("DUNST_BODY",      body);
+                                safe_setenv("DUNST_ICON_PATH", icon);
+                                safe_setenv("DUNST_URGENCY",   urgency);
+                                safe_setenv("DUNST_ID",        n_id);
+                                safe_setenv("DUNST_PROGRESS",  n_progress);
+                                safe_setenv("DUNST_CATEGORY",  n->category);
+                                safe_setenv("DUNST_STACK_TAG", n->stack_tag);
+                                safe_setenv("DUNST_URLS",      n->urls);
+                                safe_setenv("DUNST_TIMEOUT",   n_timeout);
+                                safe_setenv("DUNST_TIMESTAMP", n_timestamp);
```

Comment:
```suggestion
                                safe_setenv("DUNST_TIMEOUT",   n_timeout/1000);
                                safe_setenv("DUNST_TIMESTAMP", n_timestamp/1000);
```

Let's return the times in the same format as provided.

Reply:
Those are actually strings. I'll go ahead and replace the ints.